src/client/mod.rs

```diff
@@ -143,35 +138,27 @@ impl<'a> ClientBuilder<'a> {
     ///
     /// [`Http`]: crate::http::Http
     pub fn new_with_http(http: Http) -> Self {
-        let mut c = Self::_new();
-        c.http = Some(http);
-        c
+        Self::_new(http)
     }
 
     /// Sets a token for the bot. If the token is not prefixed "Bot ",
     /// this method will automatically do so.
     pub fn token(mut self, token: impl AsRef<str>) -> Self {
-        let token = token.as_ref().trim();
-
-        let token =
-            if token.starts_with("Bot ") { token.to_string() } else { format!("Bot {}", token) };
-
-        self.token = Some(token.clone());
-
-        self.http = Some(Http::new_with_token(&token));
+        self.http = Http::new_with_token(token.as_ref());
 
         self
     }
 
+    pub fn get_token(&self) -> &str {
```

Comment:
There should be some documentation here.
```suggestion
    /// Gets the current token used for the [`Http`] client.
    ///
    /// [`Http`]: crate::http::Http 
    pub fn get_token(&self) -> &str {
```